contrib/babelfishpg_tsql/src/properties.c

```diff
@@ -143,23 +146,115 @@ Datum connectionproperty(PG_FUNCTION_ARGS) {
 	PG_RETURN_BYTEA_P(convertVarcharToSQLVariantByteA(vch, PG_GET_COLLATION()));
 }
 
-void* get_servername_helper()
+static VarChar *
+get_servername_helper()
 {
-	StringInfoData temp;
-    void* info;
+	StringInfoData	temp;
+	void*		info;
 
     initStringInfo(&temp);
     appendStringInfoString(&temp, bbf_servername);
 
     info = tsql_varchar_input(temp.data, temp.len, -1);
     pfree(temp.data);
-    return info;
+    return (VarChar *)info;
+}
+
+static char *
+ProcessVersionNumber(const char* inputString)
+{
+	int		part = 0;
+	const char	*product_version;
+	char		*copy_version_number = malloc(sizeof(product_version));
+	char		*token;
+
+	product_version = GetConfigOption("babelfishpg_tds.product_version", true, false);
+	strcpy(copy_version_number,product_version);
+	PG_TRY();
+	{
+		for (token = strtok(copy_version_number, "."); token; token = strtok(NULL, "."))
+		{ 
+			if(part == 0 && strcasecmp(inputString,"product_major_version") == 0)
+				return token;
+			else if(part == 1 && strcasecmp(inputString,"product_minor_version") == 0)
+				return token;
+	
+			part++;
+		}
+	}
+	PG_FINALLY();
+	{
+		free(copy_version_number);
+	}
+	PG_END_TRY();
+	
+	return token;
+}
+
+static VarChar *
+get_product_major_version_helper()
+{
+	StringInfoData	temp; 
+	void		*info;
+	const char	*product_version;
+
+	product_version = GetConfigOption("babelfishpg_tds.product_version", true, false);
+    initStringInfo(&temp);
+	if(strcasecmp(product_version,"default") == 0)
+		appendStringInfoString(&temp, BABEL_COMPATIBILITY_MAJOR_VERSION);
+    else
+		appendStringInfoString(&temp, ProcessVersionNumber("product_major_version"));
+		
+    info = tsql_varchar_input(temp.data, temp.len, -1);
+    pfree(temp.data);
+    return (VarChar *)info;
+}
+
+static VarChar *
+get_product_minor_version_helper()
```

Comment:
`get_product_major_version_helper()` and `get_product_minor_version_helper()` are almost duplicate. We can just use `get_product_version_helper(int idx)` to do the job, e.g.,
```
idx >=0 --> return idx-th part
idx == -1 --> return the full version number
```